ocean_py/agents/agent.py

```diff
@@ -0,0 +1,91 @@
+"""
+    Agent class to provide basic functionality for all Ocean Agents
+"""
+
+import secrets
+
+from squid_py.ddo import DDO
+from squid_py.didresolver import DIDResolver
+from squid_py.did import id_to_did
+
+class Agent():
+    def __init__(self, ocean, **kwargs):
+        """init the Agent with a connection client and optional DID"""
+        self._ocean = ocean
+        self._did = kwargs.get('did', None)
+        self._ddo = None
+        # if DID then try to load in the linked DDO
+        if self._did:
+            self._ddo = self._resolve_did_to_ddo(self._did)
+
+    def register_url(self, name, endpoint, account, did=None):
+        """
+        Register this agent on the block chain
+        :param did: DID of the agent to register
+        :param name: name of the service endpoint to register
+        :param endpoint: URL of the agents service to register
+        :param account: Ethereum account to use as the owner of the DID->DDO registration
+        :return private password of the signed DDO
+        """
+
+        if did is None:
+            # if no did then we need to create a new one
+            did = id_to_did(secrets.token_hex(32))
+
+        # create a new DDO
+        ddo = DDO(did)
+        # add a signature
+        private_password = ddo.add_signature()
+        # add the service endpoint with the meta data
+        ddo.add_service(name, endpoint)
+        # add the static proof
+        ddo.add_proof(0, private_password)
+        if self.register_ddo(did, ddo, account):
+            # save this to the object once the registration has occured
```

Comment:
register_ddo will return a transaction receipt (in the happy path) -or- it will throw an exception.
If it fails do we want to catch the range of exceptions -or- propogate the exception?
Basically the "return None" is never actually expected (when things work correctly), right?

Reply:
The keeper.register method in squid-py can return None, if the consensus takes too long and web3 times out waiting for a receipt to the block chain transaction. I need to think of the best way to handle this, or maybe raise an exception when this occurs